Recipe collection

kawrou opened the pull request:
Recipe collection should be ready. But will need changing because of update to state sharing in app. 
Creating a draft pull request for now.

kawrou committed:
Started writing tests for RecipeCollection page

kawrou committed:
WIP: test failing

kawrou committed:
Fix: Test. Solution: await waitFor()

kawrou committed:
Wrote test for navigating to recipe page

kawrou committed:
Refactor: Made a custom hook useFetchRecipe and wrote test

kawrou committed:
Fixed test: RecipeCollection page navigates to login. Tested: useFetchRecipe Hook

kawrou committed:
Changed the conditional rendering

kawrou committed:
Merge branch 'main' into recipe_collection
Merging latest main

kawrou committed:
Fix: Issue with mapping over undefined recipe variable

kawrou committed:
Test: New test cases. Comments on dubious test cases

kawrou committed:
Merge branch 'main' into recipe_collection

Mergine latest changes to main From Matt's recipe page

kawrou committed:
Fix: RecipeCollection Page test

kawrou committed:
Fix: useFetchRecipe tests

kawrou committed:
Merge branch 'main' into recipe_collection

kawrou committed:
Found more broken tests after pull. Fixing

kawrou committed:
Fixed RecipeScraper test

kawrou commented:
A few of the tests are broken. I've fixed:
RecipeCollection Page
useFetchRecipe
RecipeScraper

Need to fix other tests on different branch to not hold up this PR

kawrou committed:
Fixed recipes.test for services/recipes

kawrou commented:
Fixed recipes.test for services/recipes

mattwsheps commented on frontend/src/App.jsx:
```diff
@@ -94,7 +94,18 @@ const App = () => {
         />
-        <Route path="/recipecollection" element={<RecipeCollection />} />
+        <Route
+          path="/recipecollection"
```
Change path to '/myrecipes'

kawrou replied:
> Change path to '/myrecipes'

Done!

mattwsheps reviewed (commented):
Looks good other than potential path name change and maybe a name change for RecipeCollection.jsx to MyRecipesPage.jsx?

kawrou committed:
Fix: HomePage test. TDD need to finish HomePage

kawrou committed:
Refactored tests

kawrou committed:
Renamed RecipeCollection - updated path in App

kawrou committed:
Added some imports from vitest to top of files

kawrou committed:
Moved code: services/getRecipes -> services/recipes. Updated uses in useFetchRecipe hook

kawrou commented:
Ok, I think I'm really finished with it now. 
I've renamed RecipeCollection
I've deleted GetRecipes service as deprecated.
The useFetchRecipes custom hook will now use the proper function inside of services/recipes

I've fixed the test in HomePage but HomePage needs some work! 
It's just to verify that a user would be navigated to a login or createRecipe page.

gabrielaehrenbrink reviewed (approved):
Looks good!
